### ISSUE TITLE
wrong autocorrect when autodetecting file names

### ISSUE THREAD

**Issue by JosephBARBIERDARNAL** [posit-dev/positron#5193]
#### Positron and OS details:

Positron Version: 2024.08.0 (Universal) build 48
Code - OSS Version: 1.91.0
Commit: ed616b36655fb24d116108bdd833f1321704315b
Date: 2024-08-19T04:26:51.868Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.6.0

#### Interpreter details:

Python 3.11.5

## Describe the issue:

When the editor suggests a filename, it adds its full name to where my cursor is, instead of replacing what I've already written, which makes it unusable for accepting suggestions.

There are several cases that I've recorded for simplicity's sake, as they're not easy to describe in writing.

Context: I want to open a file called `countries.geojson`, either in the `data/` directory (case 1) or in the root directory (cases 2 and 3).

- case 1 (does not work as expected)

![Image](https://github.com/user-attachments/assets/6144ba77-990a-48a8-bcfc-f55e58b7610c)

- case 2 (does work as expected)

![Image](https://github.com/user-attachments/assets/a8b77f86-df62-45af-9e35-bb7350447899)

- case 3 (does not work as expected)

![Image](https://github.com/user-attachments/assets/bbb4db96-c156-43c0-85f9-8b048a410a8c)

_Note: I haven't managed to reproduce it in VS code or in an R environment._

## Steps to reproduce the issue:

```python
import geopandas as gpd
from shapely.geometry import shape

geojson_data = {
    "type": "FeatureCollection",
    "crs": {"type": "name", "properties": {"name": "urn:ogc:def:crs:OGC:1.3:CRS84"}},
    "features": [
        {
            "type": "Feature",
            "properties": {"name": "Fiji"},
            "geometry": {
                "type": "MultiPolygon",
                "coordinates": [
                    [
                        [
                            [180.0, -16.067132663642447],
                            [180.0, -16.555216566639196],
                            [180.0, -16.555216566639196],
                            [180.0, -16.555216566639196],
                        ]
                    ],
                    [
                        [
                            [178.12557, -17.50481],
                            [178.3736, -17.33992],
                            [178.3736, -17.33992],
                            [178.3736, -17.33992],
                        ]
                    ],
                ],
            },
        },
        {
            "type": "Feature",
            "properties": {"name": "Tanzania"},
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [33.903711197104528, -0.95],
                        [34.07262, -1.05982],
                        [37.69869, -3.09699],
                        [37.69869, -3.09699],
                    ]
                ],
            },
        },
    ],
}


features = geojson_data["features"]
geometries = [shape(feature["geometry"]) for feature in features]
properties = [feature["properties"] for feature in features]
gdf = gpd.GeoDataFrame(properties, geometry=geometries)
gdf.to_file("countries.geojson", driver="GeoJSON")

# try writing "countries.geojson" here
gpd.read_file()
```

## Expected or desired behavior:

Instead of `data/countries.countries.geojson`, I would like the replacement to be `data/countries.geojson`.

**Comment by JosephBARBIERDARNAL** [comment]
@juliasilge Do you want me to open an exit on the Jedi repo? Or will Positron handle the problem directly?

**Comment by juliasilge** [comment]
@JosephBARBIERDARNAL We will likely want to open up an issue over there, but it would be better to show the bug in a more straightforward usage than Positron, like using Jedi in VS Code (change the setting "Python: Language Server" from default to Jedi) or emacs or similar. I did just confirm I see the same bug in VS Code when using Jedi. No expectation but if you are up for it, that would be most welcome!

**Comment by jthomasmock** [comment]
Added an issue + example over in Jedi repo including a screencast of the same behavior in VS Code when using Jedi as the LSP.

**Comment by jthomasmock** [comment]
Hmm for what it's worth, I got a response that this may be a problem outside of Jedi.

I did try the Path Intellisense and Path Autocomplete extensions which seems to resolve the issue when auto-completing at the `.` -- note that I had to start using <kbd>Ctrl</kbd>+<kbd>Space</kbd> to initiate the auto-complete rather than <kbd>Tab</kbd>

**Comment by etiennebacher** [comment]
I don't know if this is the same root cause but I observed a similar issue  in R when autocomplete is applied on a filename with a dash `-`. In the example below, the autocompletion works fine when I type only `test`, but if I type `test-` and then run autocompletion the word `test` is included twice in the autocompleted path (as in cases 1 and 3 in the original post):

https://github.com/user-attachments/assets/5bc72a6b-2bd0-4b16-9fcd-231333f1128a

**Comment by juliasilge** [comment]
Thanks for the report @etiennebacher! I opened a new issue in #5623 since this is specific to the R path completions.

**Comment by seeM** [comment]
@jthomasmock even though path completions were removed from `jedi-language-server`, we are still completing paths after upgrading just because of the way we've extended their code (we had to copy-paste and edit the whole completions method).

The underlying `jedi` library handles path completions quite well, so I decided to make the relatively simple fix (#6002) instead of following `jedi-language-server` and removing path completions entirely.

I suggest we decide whether to remove path completions from our Python language server when we tackle #5345.

**Comment by jthomasmock** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/29187501?s=64&amp;u=f88ad2fe7dfa4f00ff0dd37d42dc3a4fab50eeec&amp;v=4">@jthomasmock](https://github.com/jthomasmock) even though path completions were removed from `jedi-language-server`, we are still completing paths after upgrading just because of the way we've extended their code (we had to copy-paste and edit the whole completions method).
> 
> The underlying `jedi` library handles path completions quite well, so I decided to make the relatively simple fix ([#6002](https://github.com/posit-dev/positron/pull/6002)) instead of following `jedi-language-server` and removing path completions entirely.
> 
> I suggest we decide whether to remove path completions from our Python language server when we tackle [#5345](https://github.com/posit-dev/positron/issues/5345).

Agreed -- thanks Wasim. I understand the Jedi team's choice to remove the capability, but I think our users will expect path/file completions (and accurate completions!)

**Comment by jonvanausdeln** [comment]
### Verified Fixed
Positron Version(s) : [2025.02.0-87](https://github.com/posit-dev/positron-builds/releases/tag/2025.02.0-87)
OS Version(s) : Windows 11

### Test scenario(s) 
File completions work as expected

### Link(s) to TestRail test cases run or created:
